src/fifobuffer.jl

```diff
@@ -79,10 +80,10 @@ Base.read(f::FIFOBuffer) = readavailable(f)
 Base.flush(f::FIFOBuffer) = nothing
 
 function Base.eof(f::FIFOBuffer)
-    return f.eof && f.nb == 0
+    return f.content_length >= 0 && f.content_pos >= f.content_length || f.content_length < 0 && f.nb == 0
 end
 function Base.close(f::FIFOBuffer)
-    f.eof = true
+    f.content_length = f.content_pos
```

Comment:
I actually think the correct behavior here is to do nothing:
1. If there was no `content_length`, it doesn't matter what we do
2. If there was a `content_length`, we don't want to change anything because `eof` relies on having `content_pos` >= `content_length`; we wouldn't want an early `eof` because the buffer got closed. `close` means no more writing, but doesn't preclude reading what's left in the buffer.